.circleci/config.yml

```diff
@@ -3320,6 +3392,8 @@ workflows:
               parallelism: [2]
               cluster-id-prefix: ["nightly"]
               mark: ["nightly"]
+          requires:
```

Comment:
Could you avoid these requires if you used a when clause on the nightly workflows to prevent it other than by 

    when:
      or:
        - equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
        - equal: [ api, << pipeline.trigger_source >> ]
        
And change check-if-nightly-should run to trigger the nightly to start it via the API instead of cancelling it via the API?

https://circleci.com/docs/triggers-overview/#run-a-pipeline-using-the-api

Reply:
The API doesn't have the option of targeting a specific workflow, so that means that all workflows will run again, possibly getting us into a recursive loop?

Also, `check-if-nightly-should-run` is job that is started from the `nightly` workflow itself.  It would need to get called from somewhere else, then.  So, we would have to run it as part of another workflow, I suppose?

Reply:
> The API doesn't have the option of targeting a specific workflow,

Oh.  That's a problem then.   Never mind.   They don't make it sound that way in the docs.
Unless you could put in a parameter, and then have all other workflows 'when not' on that parameter.      That just flips the problem the other way I guess as there are 10 workflows that you might need to change. :-(

> Also, check-if-nightly-should-run is job that is started from the nightly

Right, you would just include trigger-if-nightly-should-run in the default workflow.   But if you can't easily just trigger  a single workflow then this probably isn't work pursuing.

Reply:
Yeah, Danny and I already had discussed this in my EE pull request.

https://github.com/determined-ai/determined-ee/pull/988#discussion_r1303579958